src/yarpdatadumper/main.cpp

```diff
@@ -112,39 +111,58 @@ class DumpImage : public DumpObj
 
     const string toFile(const string &dirName, unsigned int cnt) override
     {
-        file::image_fileformat format;
+        file::image_fileformat fileformat = file::FORMAT_NULL;
         string ext;
 
         int code=p->getPixelCode();
-        if (code==VOCAB_PIXEL_MONO_FLOAT)
-        {
-            format=file::FORMAT_NUMERIC;
-            ext=".float";
-        }
-        else if (code==VOCAB_PIXEL_MONO)
+        switch (code)
         {
-            format=file::FORMAT_PGM;
-            ext=".pgm";
-        }
-        else if (dump_format==DumpFormat::image_jpg)
-        {
-            format=file::FORMAT_JPG;
-            ext=".jpg";
-        }
-        else if (dump_format==DumpFormat::image_png)
-        {
-            format=file::FORMAT_PNG;
-            ext=".png";
-        }
-        else
-        {
-            format=file::FORMAT_PPM;
-            ext=".ppm";
+            //depth images
+            case VOCAB_PIXEL_MONO_FLOAT:
+                if (m_dump_format == DumpFormat::depth)
+                {
+                    fileformat = file::FORMAT_NUMERIC;
+                    ext = ".float";
+                }
+                else if (m_dump_format == DumpFormat::depth_compressed)
+                {
+                    fileformat = file::FORMAT_NUMERIC_COMPRESSED;
+                    ext = ".floatzip";
+                }
+            break;
+
+            //grayscale images
+            case VOCAB_PIXEL_MONO:
+                if (1)
```

Comment:
This `if` can be actually removed.

Reply:
That one was intentional..
If (1) is generally nice: it does not generate warnings, is optimized out, it preserves the same indentation of the following sections, it reminds the developer that further features should be added in that specific point to respect a design pattern (in this case: do we want to add the possibility of saving grayscale images as color images? Not for now, but in the feature...)

However:

After some investigation, I noticed that png format supports grayscale, and no additional code is needed. So that piece of code has been replaced with real decision branch.